### PR TITLE
Make comparison of polynomials error if parent check fails

### PR DESCRIPTION
**Pull request by joschmitt**
One way to resolve https://github.com/oscar-system/Oscar.jl/issues/4107 . Let's see what CI says.

### CODE REVIEW

File: src/MPoly.jl [hunk 1]
```diff
@@ -1515,3 +1515,14 @@ multivariate polynomial ring.
...
+
+###############################################################################
+#
+#   Comparisons
+#
+###############################################################################
+
+function ==(x::MPolyRingElem, y::MPolyRingElem)
+   check_parent(x, y, true)
+   error("Polynomials cannot be compared")
```
Comment (fingolfin) [inline src/MPoly.jl:1527]:
```suggestion
   error("Polynomial comparison not implemented")
```
would be more accurate

Comment (joschmitt) [reply src/MPoly.jl:1527]: True.

File: src/MPoly.jl [hunk 2]
```diff
@@ -1515,3 +1515,14 @@ multivariate polynomial ring.
...
+
+###############################################################################
+#
+#   Comparisons
+#
+###############################################################################
+
+function ==(x::MPolyRingElem, y::MPolyRingElem)
+   check_parent(x, y, true)
+   error("Polynomial comparison not implemented")
```
Comment (fingolfin) [inline src/MPoly.jl:1527]: How about
```suggestion
   return is_zero(x - y)
```

**Review by fingolfin — APPROVED** [review]
I am fine with this in principle, although I am not sure how useful it is in general. But there definitely is no harm.

**Comment by lgoettgens** [comment]
I would prefer to just implement `==(::AbstractAlgebra.SetElem, ::AbstractAlgebra.SetElem) = error("== is not implemented for the given types")` if we wanna have this

**Comment by lgoettgens** [comment]
> I would prefer to just implement `==(::AbstractAlgebra.SetElem, ::AbstractAlgebra.SetElem) = error("== is not implemented for the given types")` if we wanna have this

Triage agreed with this idea

**Comment by joschmitt** [comment]
I close this now. I don't know what the current status of the general problem is, but this seemed not to be the desired solution any way.

**Pull request closed without merging**